### PR TITLE
Fixes #1365 - Always export service ports in app.ports field

### LINKED ISSUE #1365: servicePorts are not copied into ports
I pushed the 2 following applications to my cluster:
```json
{
  "container": {
    "type": "DOCKER",
    "docker": {
      "image": "nginx",
      "network": "BRIDGE",
      "portMappings": [
        { "containerPort": 80, "hostPort": 0, "servicePort": 0, "protocol": "tcp" },
        { "containerPort": 80, "hostPort": 0, "servicePort": 0, "protocol": "tcp" }
      ]
    }
  },
  "id": "network2",
  "instances": 1,
  "cpus": 0.1,
  "mem": 64
}
```

```json
{
  "container": {
    "type": "DOCKER",
    "docker": {
      "image": "nginx",
      "network": "BRIDGE",
      "portMappings": [
        { "containerPort": 80, "hostPort": 0, "servicePort": 80, "protocol": "tcp" },
        { "containerPort": 80, "hostPort": 0, "servicePort": 81, "protocol": "tcp" }
      ]
    }
  },
  "id": "network4",
  "instances": 1,
  "cpus": 0.1,
  "mem": 64
}
```

But the API returns me the following:
```bash
$ curl -s http://<marathon ip>:8080/v2/apps | jq '.apps[] | {ports: .ports, id: .id, portMappings: .container.docker.portMappings }'
````
```json
{
  "portMappings": [
    {
      "protocol": "tcp",
      "servicePort": 10000,
      "hostPort": 0,
      "containerPort": 80
    },
    {
      "protocol": "tcp",
      "servicePort": 10001,
      "hostPort": 0,
      "containerPort": 80
    }
  ],
  "id": "/network2",
  "ports": [
    10000,
    10001
  ]
}
{
  "portMappings": [
    {
      "protocol": "tcp",
      "servicePort": 80,
      "hostPort": 0,
      "containerPort": 80
    },
    {
      "protocol": "tcp",
      "servicePort": 81,
      "hostPort": 0,
      "containerPort": 80
    }
  ],
  "id": "/network4",
  "ports": [
    0
  ]
}
```

In the second case, I expect to have `ports: [80,81]`.
Am I wrong?

### PR DESCRIPTION
**Pull request by kolloch**
...even if there are no dynamic app ports to assign.

Also:

* cleanup some methods without side effects that had a () call convention.
* improve documentation about --local_port_min and --local_port_max

### CODE REVIEW

File: src/test/scala/mesosphere/marathon/state/GroupManagerTest.scala [hunk 1]
```diff
@@ -57,17 +57,40 @@ class GroupManagerTest extends TestKit(ActorSystem("System")) with MockitoSugar
     val group = Group(PathId.empty, Set(
       AppDefinition("/app1".toPath, ports = Seq(), container = Some(container))
     ))
-    val update = manager(10, 20).assignDynamicAppPort(Group.empty, group)
+    val update = manager(minServicePort = 10, maxServicePort = 20).assignDynamicServicePorts(Group.empty, group)
     update.transitiveApps.filter(_.hasDynamicPort) should be ('empty)
     update.transitiveApps.flatMap(_.ports.filter(x => x >= 10 && x <= 20)) should have size 2
   }
 
+  // Regression test for #1365
```
Comment (aquamatthias) [inline src/test/scala/mesosphere/marathon/state/GroupManagerTest.scala]: Could you add tests for: 
-only ports (no container)
-mixed port and container

File: src/main/scala/mesosphere/marathon/state/GroupManager.scala [hunk 2]
```diff
@@ -225,7 +225,12 @@ class GroupManager @Singleton @Inject() (
     }
 
     val dynamicApps: Set[AppDefinition] =
-      to.transitiveApps.filter(_.hasDynamicPort).map(assignPorts)
+      to.transitiveApps.map {
+        case app if app.hasDynamicPort => assignPorts(app)
+        case app =>
```
Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/state/GroupManager.scala]: Do we need to migrate the old applications?

File: src/test/scala/mesosphere/marathon/state/GroupManagerTest.scala (hunk 1 above)
Comment (kolloch) [inline src/test/scala/mesosphere/marathon/state/GroupManagerTest.scala]: good point

File: src/main/scala/mesosphere/marathon/state/GroupManager.scala (hunk 2 above)
Comment (kolloch) [inline src/main/scala/mesosphere/marathon/state/GroupManager.scala]: I guess so :)

**Comment by kolloch** [comment]
I implemented your suggestions

**Comment by aquamatthias** [comment]
Can you please rebase?

**Comment by kolloch** [comment]
rebased

**Comment by aquamatthias** [comment]
:+1:

**Pull request merged**